### ISSUE TITLE
`LineReceiver` API documentation should indicate when delimiter is stripped etc

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4542]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#4542|
|Type|enhancement|
|Created|2010-07-04 15:08:55Z|
|Branch|https://github.com/twisted/twisted/tree/linereceiver-docs-4542-2|

`LineReceiver.lineReceived` is called with a line sans delimiter. `LineReceiver.sendLine` will add the delimiter to whatever is passed to it. These facts should be included in the API documentation for `LineReceiver`.

<details><summary>Searchable metadata</summary>

```
trac-id__4542 4542
type__enhancement enhancement
reporter__jml jml
priority__normal normal
milestone__ 
branch__branches_linereceiver_docs_4542_2 branches/linereceiver-docs-4542-2
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1278256135000000 1278256135000000
changetime__1279773564000000 1279773564000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29607]) Branching to 'linereceiver-docs-4542'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Please review

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to @exarkun|
|-|-|

Thanks for working on this, exarkun. Some review feedback:

 * copyright bump
 * the two sendLines need @type markup for the 'line' parameter.
 * "return self.transport.writeSequence((line,self.delimiter))" ===> "return self.transport.writeSequence((line, self.delimiter))"
 * IntNStringReceiver.stringReceived has the 'msg' parameter but adds epytext markup for 'string' (the rest of the stringReceived methods in the file pass 'string'). [ this doesn't cause a pydoctor error :-( ]

side note:
 * SafeNetstringReceiver is deprecated in the docstring but not with any code support

Other than that, this looks good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [29783]) Branching to 'linereceiver-docs-4542-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun removed owner|
|-|-|

Some wicked conflicts with the netstring branch that just landed.  I merged forward and resolved them, ended up changing a few more things than were mentioned in the review, so another check would be nice.

Everything mentioned should be fixed now.  There's [#4546](https://github.com/twisted/twisted/issues/4546) for `SafeNetstringReceiver`.

Thanks for the review!

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|TimAllen set owner to @exarkun|
|-|-|

Looks good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/linereceiver-docs-4542-2)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [29883]) Merge linereceiver-docs-4542-2

Author: exarkun
Reviewer: jesstess, screwtape
Fixes: [#4542](https://github.com/twisted/twisted/issues/4542)

Make the single parameter to the `sendString` methods of classes in
`twisted.protocols.basic` consistently named and document it.  Do the
same for the parameter to the `stringReceived` and `lineReceived`
methods as well.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
